brushfire-scalding/src/main/scala/com/stripe/brushfire/scalding/Trainer.scala

```diff
@@ -339,19 +363,26 @@ case class Trainer[K: Ordering, V, T: Monoid](
 object Trainer {
   val MaxReducers = 20
 
-  def apply[K: Ordering, V, T: Monoid](trainingData: TypedPipe[Instance[K, V, T]], sampler: Sampler[K]): Trainer[K, V, T] = {
-    val empty = 0.until(sampler.numTrees).map { treeIndex => (treeIndex, Tree.singleton[K, V, T](Monoid.zero)) }
+  def apply[M, K: Ordering, V, T: Monoid, A](
+      trainingData: TypedPipe[Instance[M, Map[K, V], T]],
+      annotator: Annotator[M, A],
+      sampler: Sampler[M, K],
+      id: M => String): Trainer[M, K, V, T, A] = {
+    implicit val annotationSemigroup = annotator.monoid
```

Comment:
Yeah, I mean - the only reason for that is there's no other natural place for the Monoid[T] to live... I agree it's pretty arbitrary but I can live with it.